lib/awaitify-stream.js

```diff
@@ -111,6 +116,7 @@ function addReaderFunctions(readStream, obj) {
                 // wait for more data to be available, or the end of the stream.
                 readStream.once('readable', read);
                 readStream.once('end', read);
+                readStream.once('error', read);
```

Comment:
It seems like this is relying on the error event handler at https://github.com/hashtagchris/node-awaitify-stream/blob/master/lib/awaitify-stream.js#L79 being called first and setting `error`, followed by `read`.

Can we rely on that? Should `read` have an `err` parameter in case it gets called first?

Reply:
We can rely on that since https://github.com/hashtagchris/node-awaitify-stream/blob/master/lib/awaitify-stream.js#L79 is registered first and:

> By default, event listeners are invoked in the order they are added. 

https://nodejs.org/dist/latest-v8.x/docs/api/events.html#events_emitter_on_eventname_listener